### PR TITLE
Add EXT_mesh_features support

### PR DESCRIPTION
**Pull request by AntanasBuk**
This PR introduces support for reading and interpreting Cesium tiles with [`EXT_mesh_features` extension](https://github.com/CesiumGS/glTF/blob/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features/README.md). Support only extends to tilesets that hold [Feature ID information by Vertex](https://github.com/CesiumGS/glTF/blob/3d-tiles-next/extensions/2.0/Vendor/EXT_mesh_features/README.md#feature-id-by-vertex) and will ignore Feature ID by Texture Coordinates.

This feature was requested here: https://github.com/iTwin/itwin-graphics-backlog/issues/425

The new behavior can be tested by:
1. Serve Cesium tileset that has `EXT_mesh_features` on localhost
2. Run display-test-app
3. Open a blank connection
4. In the integrated console enter `fdt attach reality model <path to served tileset.json>`
5. If the tileset is not visible, try pressing `Fit to View` button
6. Hover over some visible object - a tooltip with object metadata should appear.

ext_mesh_features
Model used in this image was also used in GltfReader tests.

### CODE REVIEW

File: core/frontend/src/test/tile/GltfReader.test.ts [hunk 1]
```diff
@@ -977,96 +1249,148 @@ describe("GltfReader", () => {
...
-      expect(entryCount === 4).toBe(true);
+    describe("EXT_mesh_features", () => {
+      it("feature indices should map to correct vertices", async () => {
+        const elementIdToStructuralMetadataMap = new BatchedTileIdMap(iModel);
+
+        const reader = createReader(meshFeaturesExt, elementIdToStructuralMetadataMap)!;
+        expect(reader).toBeDefined();
+
+        const result = await reader.read();
+        expect(result).toBeDefined();
```
Comment (pmconne) [inline core/frontend/src/test/tile/GltfReader.test.ts:1354]: A more robust test would verify that the features are actually present in `result`. Your test currently only verifies that the reader parsed them.

Comment (AntanasBuk) [reply core/frontend/src/test/tile/GltfReader.test.ts:1354]: Extended the tests case.

**Comment by pmconne** [comment]
@Mergifyio backport release/4.11.x

**Pull request merged**